### PR TITLE
[bug](jdbc) fix jdbc external table with char type length error

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes
Now have test pg and oracle with char(100), if data='abc'
but read string data length is 100, so need trim extral spaces

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.88 seconds
 load time: 633 seconds
 storage size: 17123641493 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221227064449_clickbench_pr_69196.html

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 1]
```diff
@@ -417,8 +417,19 @@ Status JdbcConnector::_convert_column_data(JNIEnv* env, jobject jobj,
         M(TYPE_FLOAT, float, vectorized::ColumnVector<vectorized::Float32>)
         M(TYPE_DOUBLE, double, vectorized::ColumnVector<vectorized::Float64>)
 #undef M
+    case TYPE_CHAR: {
+        std::string data = _jobject_to_string(env, jobj);
+        // Now have test pg and oracle with char(100), if data='abc'
+        // but read string data length is 100, so need trim extral spaces
```
Comment (morningman) [inline be/src/vec/exec/vjdbc_connector.cpp:423]:
```suggestion
        // but read string data length is 100, so need trim extra spaces
```

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 2]
```diff
@@ -417,8 +417,19 @@ Status JdbcConnector::_convert_column_data(JNIEnv* env, jobject jobj,
...
+        if ((_conn_param.table_type == TOdbcTableType::POSTGRESQL) ||
+            (_conn_param.table_type == TOdbcTableType::ORACLE)) {
+            data = data.erase(data.find_last_not_of(' ') + 1);
```
Comment (morningman) [inline be/src/vec/exec/vjdbc_connector.cpp:426]: What if there are spaces in user data? like `abc    `?

File: be/src/vec/exec/vjdbc_connector.cpp (hunk 1 above)
Comment (morningman) [inline be/src/vec/exec/vjdbc_connector.cpp:423]:
```suggestion
        // but read string data length is 100, so need trim extra spaces
```

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**